event_response_tool/layer_loss_duplicator/duplicate_layer_loss.py

```diff
@@ -0,0 +1,348 @@
+                elif layer.type == "FilterLayer":
+                    # Transform any loss sets that the Filter Layer may have
+                    loss_sets = []
+
+                    for loss_set in layer.loss_sets:
+                        transformed = self.transform_loss_set(loss_set)
+                        if transformed:
+                            # Add transformed loss set to cache
+                            self.loss_set_mapping[
+                                loss_set.id
+                            ] = transformed.id
+                            loss_sets.append(transformed)
+
+                    layer = self.replace_filter_layer(loss_sets)
+                else:
+                    # Transform the loss sets in the loss set list in-place
+                    loss_sets = []
+
+                    for index, loss_set in enumerate(layer.loss_sets):
+                        # Skip YELTs and parametric loss sets
+                        if (
+                            loss_set.type == "ParametricLossSet"
+                            or loss_set.type == "YELTLossSet"
+                        ):
+                            alert.warning(
+                                f"Encountered {loss_set.type} {loss_set.id}, skipping transformation"
+                            )
+                            loss_sets.append(loss_set)
+
+                        # Check if loss set is already processed
+                        elif loss_set.id in self.loss_set_mapping:
+                            loss_sets.append(
+                                analyzere.LossSet.retrieve(
+                                    self.loss_set_mapping[loss_set.id]
+                                )
+                            )
+
+                        # If unknown loss set, transform it.
+                        else:
+                            transformed = self.transform_loss_set(loss_set)
+                            if transformed:
+                                self.loss_set_mapping[
+                                    loss_set.id
+                                ] = transformed.id
+                                loss_sets.append(transformed)
+
+                    layer.loss_sets = loss_sets
+            except Exception as e:
+                alert.exception(f"Unable to process layer: {e}")
+        return layer
+
+    def process_layer(self, layer_uuid):
+        try:
+            input_layer = analyzere.LayerView.retrieve(layer_uuid)
+            if input_layer:
+                old_layer_view = input_layer
+
+                # Update LayerView and compute EL of original and modified LayerView
+                new_layer_view = analyzere.LayerView(
+                    analysis_profile=input_layer.analysis_profile,
+                    layer=self.modify_layer(input_layer),
+                ).save()
+
+                # Compute share-applied EL by creating a temporary PortfolioView
```

Comment:
comments could be clarified that this is preparation for EL computation, not the EL computation